src/ParallelAlgorithms/Interpolation/Actions/ElementReceiveInterpPoints.hpp

```diff
@@ -31,12 +31,12 @@ namespace Actions {
 template <typename InterpolationTargetTag>
 struct ElementReceiveInterpPoints {
   template <typename ParallelComponent, typename DbTags, typename Metavariables,
-            typename ArrayIndex>
+            typename ArrayIndex, size_t Dim>
   static void apply(
       db::DataBox<DbTags>& box,
       const Parallel::GlobalCache<Metavariables>& /*cache*/,
       const ArrayIndex& /*array_index*/,
-      tnsr::I<DataVector, Metavariables::volume_dim,
+      tnsr::I<DataVector, Dim,
```

Comment:
We've generally found "reaching up" into the metavariables to be fairly brittle and tricky. Considering we deduce the dim pretty much everywhere in the code I saw no reason to avoid doing that here. I've also sometimes had issues with GCC not being able to retrieve `Metavaraibles::volume_dim` and those were extremely difficult to debug, so I feel like suggesting that as the "default" is not really a good thing.

Reply:
I agree in general we shouldn't reach up, but I thought the `volume_dim` was an exception since it's so ubiquitous in our metavars. We also do this a lot throughout the code for the dim, so we aren't really consistent. But if it's causing issues with GCC then passing in the Dim explicitly is ok